integrations/oci/vault-health/src/main/java/io/helidon/integrations/oci/vault/health/OciVaultHealthCheck.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.vault.health;
+
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.helidon.config.Config;
+import io.helidon.health.common.BuiltInHealthCheck;
+import io.helidon.integrations.common.rest.ApiOptionalResponse;
+import io.helidon.integrations.oci.vault.GetVault;
+import io.helidon.integrations.oci.vault.OciVault;
+import io.helidon.integrations.oci.vault.OciVaultRx;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Liveness;
+
+import static io.helidon.common.http.Http.Status.OK_200;
+
+/**
+ * Liveness check for an OCI's Vault. Reads OCI properties from '~/.oci/config'.
+ */
+@Liveness
+@ApplicationScoped  // this will be ignored if not within CDI
+@BuiltInHealthCheck
```

Comment:
Built in health checks should only be used for those that should be disabled by `mp.health.disable-default-procedures` - I do not think this belongs to the list.
On the other hand you did the `BuiltInHealthCheck` annotation, so maybe my expectation is wrong here...

Reply:
That's interesting. I have thought that annotation would apply to any health check that isn't application defined, but I can see why we may want to separate these. If dropped here, it should probably be dropped for object storage as well.